src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -254,9 +303,20 @@ def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.devic
         step_ratio = self.config.num_train_timesteps // self.num_inference_steps
```

Comment:
`step_ratio` must not be floored to int here. It should be kept as float here for `trailing` to space more accurately below. See the paper, we deliberately not use floor operation to calculate interval.

So this is tricky. Unlike `leading` which always has the same integer spacing. `trailing` and `linspace` may have uneven integer spacing.

We need to change `step()` function to also make sure it supports this.

Overrall I think we should support user define any custom sampling timestep such as: [0, 99, 499, 999] in the future. `step` function should not expect even spacing!

Reply:
Think we could solve this in a future PR. For now it's important that timesteps are full integer numbers I'd say

Reply:
Ok let's move to uneven step ratio for `"trailing"` - don't see a problem with this :-)